eo-maven-plugin/src/main/resources/org/eolang/maven/pre/to-java.xsl

```diff
@@ -79,7 +85,15 @@ SOFTWARE.
         </xsl:otherwise>
       </xsl:choose>
     </xsl:variable>
-    <xsl:value-of select="concat($p, eo:clean($c))"/>
+    <xsl:variable name="pre" select="concat($p, eo:clean($c))"/>
+    <xsl:choose>
+      <xsl:when test="string-length($pre)&gt;250">
+        <xsl:value-of select="concat(substring($pre, 1, 25), $alt)"/>
```

Comment:
Agreed, maybe you could introduce another naming convention for long class names to avoid possible collisions? For instance, if class name is too long you could just hash it somehow or smth like that.

Reply:
@OlesiaSub @yegor256 Hashing would be ideal solution. 
Do you have an idea how to create it in XSL language?

Reply:
@mximp take a look [https://stackoverflow.com/questions/62388458/create-an-md5-encryption-function-using-pure-xsl](url), maybe it will be helpful..
But actually, if hashing is impossible/too complicated in xsl, can't you just create _some_ names using XSL and then hash the long ones using non-xsl methods? (I'm not sure how this translator to java works, so maybe this idea is useless or impossible to implement too :) )

Reply:
@OlesiaSub thanks for the hint.. That's exactly what I'm talking about: there is no simple and elegant solution to implement it. The suggestion by the link is to use integration with Java which sounds overly complex.
Actually combination of file name, line and position of an object will give unique short name.

Reply:
@mximp ok got it. can I ask Yegor to merge it if the Codacy check doesn't pass though?

Reply:
@OlesiaSub I believe we ignore Codacy checks.
@yegor256 do you agree with uniqueness by name+line+position?

Reply:
@mximp @OlesiaSub it's possible to marry Java and XSL. You can implement a `Shift` in Java, where you will inject MD5 hashes into the right places. See, how it works here: https://github.com/polystat/far/blob/master/src/main/java/org/polystat/far/FaR.java#L110-L134

Reply:
@yegor256 I've just checked suggested example. This implies that we will need to make transformation after all Java code is created. And we will just do text replacement.
I still suggest initial solution with line/position. It guaranties uniqueness across the file and also preserves some meaning of the name. We also keep original object name in `@XmirObject` annotation.
What do you think?

Reply:
@mximp I don't mind at all, as long as names are unique

Reply:
@yegor256 then please merge